File: lib/models/claude_session.dart
class ClaudeSession {
  String? relaySessionId;
  bool isOffline = false;
  final List<Map<String, String>> messages;
  String? lastQuery;
  String? lastAnswer;

  ClaudeSession() : messages = [];

  void addUser(String text) => messages.add({'role': 'user', 'content': text});
  void addAssistant(String text) =>
      messages.add({'role': 'assistant', 'content': text});

  void reset() {
    relaySessionId = null;
    messages.clear();
    isOffline = false;
    lastQuery = null;
    lastAnswer = null;
  }

  static const int maxTurns = 20;
}

File: COGOS/Session/ClaudeSession.swift
import Foundation

/// Persistent conversation context. Ports `lib/models/claude_session.dart`.
final class ClaudeSession {
    static let maxTurns = 20

    struct Message {
        let role: String  // "user" or "assistant"
        let content: String
    }

    var messages: [Message] = []
    var lastQuery: String?
    var lastAnswer: String?

    func addUser(_ text: String) { messages.append(.init(role: "user", content: text)) }
    func addAssistant(_ text: String) { messages.append(.init(role: "assistant", content: text)) }

    func reset() {
        messages.removeAll()
        lastQuery = nil
        lastAnswer = nil
    }
}
